controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -323,6 +324,16 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				},
 			),
 		).
+		// Watcher for csiDriver
+		Watches(
+			&csiopv1a1.Driver{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      util.RbdDriverName,
+					Namespace: r.OperatorNamespace,
+				},
```

Comment:
This does not do what you want it do to. Watch uses the passed struct only for getting the GVK out of it using the schema mappings. All of the information you pass here will just get ignored and the watch will be for all Driver CRs.

What you need is a predicate that select on an object key. I think we already implements such predicate in the predicate utils file